boards/BMS/Src/App/states/App_AllStates.c

```diff
@@ -155,6 +156,23 @@ bool App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
         status = false;
         App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
     }
+    // Checks if the charger has thrown a fault, the disabling of the charger, etc is done with ChargeStateRunOnExit
+    if (App_Charger_IsConnected(charger))
+    {
+        if (App_Charger_HasFaulted(charger))
+        {
+            status = false;
+            App_CanRx_DEBUG_ChargingSwitch_StartCharging_Update(false);
+            App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+        }
+    }
+    else if (App_SharedStateMachine_GetCurrentState(state_machine) == App_GetChargeState())
+    {
+        status = false;
+        App_CanTx_BMS_Faults_ChargerDisconnectedInChargeState_Set(true);
+        App_CanRx_DEBUG_ChargingSwitch_StartCharging_Update(false);
+        App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+    }
```

Comment:
Why is this check here now vs. in App_ChargeState previously? I think it makes more sense there since that's where all the other charging faults are handled

Reply:
This was part of Edwin's charging diagram, it was found to not work here at all because of how the charger sends its fault signal, basically, it sends the fault signal whenever it is not currently charging, even if there is nothing "wrong", so checking in all-states will perpetually send the BMS into fault state. 

TLDR: moved the check back to charge state